Worker.sedona

```diff
@@ -163,47 +196,48 @@ abstract internal class Worker extends Component
   **
   private void updateStatus() 
   {
+    int off_state
+    off_state = sessionConnecting ? Connecting : Disconnected
+
     if (handle == null)
-      status := Disconnected
+      status := off_state
     else
-      status := isSessionLive(handle) ? Connected : Disconnected 
+      status := isSessionLive(handle) ? Connected : off_state
   }
   
   virtual bool work()
   {
     updateStatus()
-    if (status == Disconnected)
+    if (status != Connected)
     {
       // if session failed to start within sessionTimeout seconds
-      if (handle != null && Sys.ticks()-sessionStartTime>sessionTimeout)
+      if (handle != null && sessionConnecting && Sys.ticks()-sessionStartTime>sessionTimeout)
```

Comment:
from my local testing, the retry mechanism stops working now. I think the previous logics here is complicated here, I want to do a refactoring on it. And I think besides Disconnected, Connecting and Connected, we need one more state: Timeout. So all possible states will be covered, and that will make the codes more clear.

Reply:
Sorry for that.
I had my own implementation of retry, but because you've already implemented it. I've dropped my patches.